app/src/main/kotlin/org/fossify/phone/adapters/RecentCallsAdapter.kt

```diff
@@ -177,12 +178,28 @@ class RecentCallsAdapter(
 
     private fun callContact(useSimOne: Boolean) {
         val phoneNumber = getSelectedPhoneNumber() ?: return
-        activity.callContactWithSim(phoneNumber, useSimOne)
+        val name = getSelectedName() ?: return
+
+        if (activity.config.showCallConfirmation) {
+            CallConfirmationDialog(activity, name) {
+                activity.callContactWithSim(phoneNumber, useSimOne)
+            }
+        } else {
+            activity.callContactWithSim(phoneNumber, useSimOne)
+        }
     }
 
     private fun callContact() {
         val phoneNumber = getSelectedPhoneNumber() ?: return
-        (activity as SimpleActivity).startCallIntent(phoneNumber)
+        val name = getSelectedName() ?: return
+
+        if (activity.config.showCallConfirmation) {
+            CallConfirmationDialog(activity, name) {
+                (activity as SimpleActivity).startCallIntent(phoneNumber)
+            }
+        } else {
+            (activity as SimpleActivity).startCallIntent(phoneNumber)
+        }
```

Comment:
Optional: We should move this logic to an extension function since it's also used in call history and dialpad activity